pep-0678.rst

```diff
@@ -141,24 +141,24 @@ are collecting multiple exception objects to handle together. [1]_
 Specification
 =============
 
-``BaseException`` gains a new new method ``.add_note(note: str)``. Notes are
-exposed as a tuple via the read-only attribute ``__notes__``, and appear in
-the standard traceback after the exception string.  ``.add_note(note)`` replaces
-``__notes__`` with a new tuple equal to ``__notes__ + (note,)``, if ``note``
-is a string, and raises ``TypeError`` otherwise.
+``BaseException`` gains a new new method ``.add_note(note: str)``. If ``note``
+is a string, ``.add_note(note)`` appends it to the ``__notes__`` list, creating
+the attribute if it does not already exist.  If ``note`` is not a string,
+``.add_note()`` raises ``TypeError``.
 
-``del err.__notes__`` clears the contents of the ``__notes__`` attribute,
-leaving it an empty tuple as if ``.add_note()`` had never been called.  This
-allows libraries full control over the attached notes, by re-adding whatever
-they wish, without overly complicating the API or adding multiple names to
+Libraries may clear existing notes by modifying or deleting the ``__notes__``
+list, if it has been created, including clearing all notes with
+``del err.__notes__``.  This allows full control over the attached notes,
+without overly complicating the API or adding multiple names to
 ``BaseException.__dict__``.
 
 When an exception is displayed by the interpreter's builtin traceback-rendering code,
 its notes (if there are any) appear immediately after the exception message, in the order
 in which they were added, with each note starting on a new line.
 
-``BaseExceptionGroup.subgroup`` and ``BaseExceptionGroup.split`` copy the
-``__notes__`` of the original exception group to the parts.
+If ``__notes__`` has been created, ``BaseExceptionGroup.subgroup`` and
```

Comment:
Currently - only if ``__notes__`` is a sequence. I don't know yet what we should do when @warsaw assigns arbitrary objects to ``__notes__`` of exception groups and then splits them.

Reply:
I've added a paragraph to the specification which explicitly *does not specify* the behaviour when `__notes__` exists and is not a list of strings.  I'll comment with a suggestion for CPython's behaviour in these cases over on https://github.com/python/cpython/pull/31317.